shopinvader_multi_cart/services/carts.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2021 Camptocamp SA
+# @author Iván Todorovich <ivan.todorovich@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.osv import expression
+
+from odoo.addons.component.core import Component
+
+
+class CartsService(Component):
+    """Shopinvader service to manage multiple carts."""
+
+    _inherit = "shopinvader.abstract.sale.service"
+    _name = "shopinvader.carts.service"
+    _usage = "carts"
+    _expose_model = "sale.order"
+    _description = __doc__
+
+    # The following methods are 'public' and can be called from the controller.
+    # All params are untrusted so please check it by using the decorator
+    # secure params and the linked validator !
+
+    def search(self, **params):
+        """Gets every cart related to logged user
+        :param params: dict/json
+        :return: dict
+        """
+        return self._paginate_search(**params)
+
+    def get(self, _id):
+        """Gets info about given cart id.
+        :param _id: int
+        :return: dict/json
+        """
+        cart = self._get(_id)
+        return {"data": self._to_json(cart)[0]}
+
+    def select(self, _id):
+        """Selects the cart with the given id.
+        :param _id: int
+        :return: dict/json
+        """
+        cart = self._get(_id)
+        cart_data = self._to_json(cart)[0]
+        return {
+            "data": cart_data,
+            "store_cache": {"cart": cart_data},
+            "set_session": {"cart_id": cart.id},
+        }
+
+    def delete(self, _id):
+        """Deletes the cart with the given id.
+        :param _id: int
+        """
+        cart = self._get(_id)
+        cart.unlink()
+        if self.shopinvader_session.get("cart_id") == _id:
+            return self._clear_session_cart()
+        return {}
+
+    def new(self):
```

Comment:
is this really useful?

Reply:
Hmm, I think so.
I'm trying to cover this from the specs:

> I can create a new cart (just flush session cart ID then rely on next add_item to create a new one)

How would the user create a second cart without this? It's the only way to have len(carts) > 1 IMO

Reply:
yep, but as you see I didn't state an endpoint for this. What I meant, was: in the frontend I can get a new cart by simply dropping the cart ID in the session and by doing so I'll get a new cart automatically when calling `/add_item`.
Sounds good?

Reply:
Indeed. In frontend we have access to the session id of the cart so we can remove it easily

Reply:
so let's remove it

Reply:
Understood :+1: 
I've just removed it :)